src/k8s/pkg/k8sd/features/cilium/network.go

```diff
@@ -88,9 +88,12 @@ func ApplyNetwork(ctx context.Context, snap snap.Snap, cfg types.Network, _ type
 			},
 		},
 		"nodePort": map[string]any{
-			"enabled": true,
+			"enabled":           true,
+			"enableHealthCheck": false,
 		},
 		"disableEnvoyVersionCheck": true,
+		"k8sServiceHost":           cfg.GetLocalhostAddress(),
```

Comment:
Maybe we can also test that these are passed in the unit test? Only the fields that are extracted from the config.